src/Storefront/Controller/CartLineItemController.php

```diff
@@ -276,6 +276,9 @@ private function traceErrors(Cart $cart): bool
      */
     private function getLineItemArray(RequestDataBag $lineItemData): array
     {
+        if ($lineItemData->has('payload')) {
+            $lineItemData->set('payload', json_decode($lineItemData->getString('payload'), true, 512, \JSON_THROW_ON_ERROR));
```

Comment:
We should somehow make sure that only scalar values can be passed and also the amount of data should be limited somehow.

Reply:
The json_encode should guarantee that only scalars are used, since everything is converted to string, bool, int or float, or do you mean something different? 
I added a check for the byte size of the payload, if the payload is bigger than 256kb an exception is thrown, a flash message is added and the product is not added to the cart.

Reply:
Great work, thanks for the contribution. I heard this requirement many time but never found time for it :/